room/room-compiler-processing/src/main/java/androidx/room/compiler/processing/XMessager.kt

```diff
@@ -23,24 +23,47 @@ import javax.tools.Diagnostic
  */
 abstract class XMessager {
     private val watchers = mutableListOf<XMessager>()
+
     /**
-     * Prints the given [msg] to the logs while also associating it with the given [element].
+     * Prints the given [msg] to the logs while also associating it with the given [element],
+     * [annotation], and [annotationValue].
      *
      * @param kind Kind of the message
      * @param msg The actual message to report to the compiler
      * @param element The element with whom the message should be associated with
+     * @param annotation The annotation with whom the msg should be associated with
+     * @param annotationValue The annotation value with whom the msg should be associated with
      */
-    fun printMessage(kind: Diagnostic.Kind, msg: String, element: XElement? = null) {
+    @JvmOverloads
```

Comment:
In terms of API it might be better to declare the overload manually for the non-null contract of the params:
```
fun printMessage(
    kind: Diagnostic.Kind,
    msg: String,
) {
    printMsg(kind, msg, null, null, null)
}

fun printMessage(
    kind: Diagnostic.Kind,
    msg: String,
    element: XElement,
) {
    printMsg(kind, msg, element, null, null)
}

fun printMessage(
    kind: Diagnostic.Kind,
    msg: String,
    element: XElement,
    annotation: XAnnotation,
) {
    printMsg(kind, msg, element, annotation, null)
}

fun printMessage(
    kind: Diagnostic.Kind,
    msg: String,
    element: XElement,
    annotation: XAnnotation,
    annotationValue: XAnnotationValue
) {
    printMsg(kind, msg, element, annotation, annotationValue)
}

internal fun printMsg(
    kind: Diagnostic.Kind,
    msg: String,
    element: XElement? = null,
    annotation: XAnnotation? = null,
    annotationValue: XAnnotationValue? = null
) {
  // actual print
}
```